lib/saml_idp/logout_request_builder.rb

```diff
@@ -15,6 +15,10 @@ def initialize(response_id, issuer_uri, saml_slo_url, name_id, algorithm)
       self.algorithm = algorithm
     end
 
+    def reference_id
+      self.response_id
```

Comment:
This wouldn't end up making them match since it calls generate in the method call (rather than saving to variable or initializing with it [see here](https://github.com/sportngin/saml_idp/blob/master/lib/saml_idp/logout_builder.rb#L12)). You could either provide an initializer or memoize it. Also, if you look around, are these two classes the only two using `LogoutBuilder`? It seems like the whole fix could be in the parent rather than here (I'm not sure off the top of my head, would need to look at the breadth of the change)